File: Python/Open-Closed Principle/Kullanici.py
from abc import ABC, abstractmethod

class Kullanici(ABC):
    
    def __init__(self):
        self.id = None
        self.ad = None
        self.mail = None
        self.urunler = []

    @abstractmethod
    def gettoplamtutar(self, fiyat):
        pass

    def getad(self):
        return self.ad

    def geturunler(self):
        return self.urunler

    def addurun(self, urun):
        self.geturunler().append(urun)

File: Java/Open Closed Principle/Kullanici.java

import java.util.ArrayList;
import java.util.List;

abstract class Kullanici {
    protected int id;
    protected String isim;
    protected String mail;
    protected List<Urun> urunler;

    public Kullanici(){
        urunler = new ArrayList<Urun>();
    }

    public abstract double getToplamTutar(double fiyat);

    public String getIsim(){
        return isim;
    }

    public List<Urun> getUrunler(){
        return urunler;
    }

    public void addUrun(Urun urun){
        getUrunler().add(urun);
    }
    

}